habitat_baselines/il/trainers/nav_trainer.py

```diff
@@ -0,0 +1,561 @@
+        )
+
+        train_loader = DataLoader(
+            nav_dataset, batch_size=config.IL.NAV.batch_size, shuffle=True
+        )
+
+        logger.info("train_loader has %d samples" % len(nav_dataset))
+
+        q_vocab_dict, _ = nav_dataset.get_vocab_dicts()
+
+        model_kwargs = {"q_vocab": q_vocab_dict}
+        model = NavPlannerControllerModel(**model_kwargs)
+
+        planner_lossFn = MaskedNLLCriterion()
+        controller_lossFn = MaskedNLLCriterion()
+
+        metrics = NavMetric(
+            info={"split": "train"},
+            metric_names=["planner_loss", "controller_loss"],
+            log_json=os.path.join(config.OUTPUT_LOG_DIR, "train.json"),
+        )
+
+        optim = torch.optim.Adam(
+            filter(lambda p: p.requires_grad, model.parameters()),
+            lr=float(config.IL.NAV.lr),
+        )
+
+        t, epoch = 0, 1
+
+        avg_p_loss = 0.0
+        avg_c_loss = 0.0
+
+        model.train().cuda()
+
+        with TensorboardWriter(
+            config.TENSORBOARD_DIR, flush_secs=self.flush_secs
+        ) as writer:
+            while epoch <= int(config.IL.NAV.max_epochs):
+                start_time = time.time()
+                for batch in train_loader:
+                    t += 1
+
+                    (
+                        idx,
+                        questions,
+                        _,
+                        planner_img_feats,
+                        planner_actions_in,
+                        planner_actions_out,
+                        planner_action_lengths,
+                        planner_masks,
+                        controller_img_feats,
+                        controller_actions_in,
+                        planner_hidden_idx,
+                        controller_outs,
+                        controller_action_lengths,
+                        controller_masks,
+                    ) = batch
+
+                    questions = questions.cuda()
+
+                    planner_img_feats = planner_img_feats.cuda()
+                    planner_actions_in = planner_actions_in.cuda()
+                    planner_actions_out = planner_actions_out.cuda()
+                    planner_action_lengths = planner_action_lengths.cuda()
+                    planner_masks = planner_masks.cuda()
+
+                    controller_img_feats = controller_img_feats.cuda()
+                    controller_actions_in = controller_actions_in.cuda()
+                    planner_hidden_idx = planner_hidden_idx.cuda()
+                    controller_outs = controller_outs.cuda()
+                    controller_action_lengths = (
+                        controller_action_lengths.cuda()
+                    )
+                    controller_masks = controller_masks.cuda()
+
+                    (
+                        planner_action_lengths,
+                        perm_idx,
+                    ) = planner_action_lengths.sort(0, descending=True)
```

Comment:
I assume sorting is done here for using https://pytorch.org/docs/stable/nn.html?highlight=pack_padded#torch.nn.utils.rnn.pack_padded_sequence ?  If so, that doesn't requiring sorting anymore, so this is unneeded!